src/AzureRMTools.ts

```diff
@@ -91,38 +89,35 @@ export class AzureRMTools {
         context.subscriptions.push(vscode.window.registerTreeDataProvider("json-outline", jsonOutline));
         context.subscriptions.push(vscode.commands.registerCommand("extension.treeview.goto", (range: vscode.Range) => jsonOutline.goToDefinition(range)));
 
-        this.log("Extension Activated");
+        const jsonFileSubscriptionArray: vscode.Disposable[] = [];
 
-        this.logOnError(() => {
-            const jsonFileSubscriptionArray: vscode.Disposable[] = [];
+        vscode.window.onDidChangeActiveTextEditor(this.onActiveTextEditorChanged, this, jsonFileSubscriptionArray);
 
-            vscode.window.onDidChangeActiveTextEditor(this.onActiveTextEditorChanged, this, jsonFileSubscriptionArray);
+        vscode.workspace.onDidChangeConfiguration(this.loadConfiguration, this, jsonFileSubscriptionArray);
 
-            vscode.workspace.onDidChangeConfiguration(this.loadConfiguration, this, jsonFileSubscriptionArray);
+        vscode.workspace.onDidOpenTextDocument(this.onDocumentOpened, this, jsonFileSubscriptionArray);
+        vscode.workspace.onDidChangeTextDocument(this.onDocumentChanged, this, jsonFileSubscriptionArray);
+        vscode.workspace.onDidSaveTextDocument(this.onDocumentSaved, this, jsonFileSubscriptionArray);
 
-            vscode.workspace.onDidOpenTextDocument(this.onDocumentOpened, this, jsonFileSubscriptionArray);
-            vscode.workspace.onDidChangeTextDocument(this.onDocumentChanged, this, jsonFileSubscriptionArray);
-            vscode.workspace.onDidSaveTextDocument(this.onDocumentSaved, this, jsonFileSubscriptionArray);
+        this._jsonFileSubscriptions = vscode.Disposable.from(...jsonFileSubscriptionArray);
 
-            this._jsonFileSubscriptions = vscode.Disposable.from(...jsonFileSubscriptionArray);
-
-            const activeEditor: vscode.TextEditor = vscode.window.activeTextEditor;
-            if (activeEditor) {
-                this.updateDeploymentTemplate(activeEditor.document);
-            }
-        });
+        const activeEditor: vscode.TextEditor = vscode.window.activeTextEditor;
+        if (activeEditor) {
+            this.updateDeploymentTemplate(activeEditor.document);
+        }
     }
 
     public dispose(): void {
-        this.logOnError(() => {
-            if (this._deploymentTemplateFileSubscriptions) {
-                this._deploymentTemplateFileSubscriptions.dispose();
+        const me = this;
+        callWithTelemetryAndErrorHandlingSync('dispose', function (this: IActionContext) {
+            this.properties.isActivationEvent = 'true';
```

Comment:
should this suppressErrorDisplay? Might be weird if an error flashes up as you're closing VS Code